core/src/util/mod.rs

```diff
@@ -24,6 +28,52 @@ lazy_static! {
     static ref NON_ESCAPED_PERCENT: FancyRegex = FancyRegex::new(r"(?<!\\\\)%").unwrap();
 }
 
+pub(crate) fn handle_libmongosqltranslate_command(command: Document) -> Result<Document> {
+    let library = get_mongosqltranslate_library().expect("libmongosqltranslate is not available.");
+
+    let run_command_function: Symbol<
+        'static,
+        unsafe extern "C" fn(LibmongosqltranslateDataIO) -> LibmongosqltranslateDataIO,
+    > = unsafe { library.get(b"runCommand") }.expect("Failed to load runCommand symbol.");
+
+    let command_type = command.get_str("command").unwrap();
+
+    let command_bytes_vec =
+        bson::to_vec(&command).expect("Failed to serialize Document into BSON bytes.");
+
```

Comment:
Since `command` is a document, we can't use `serde_stacker`. The only other way I could find to do this would be to use [to_writer](https://docs.rs/bson/2.11.0/bson/document/struct.Document.html#method.to_writer).

Reply:
I'm not sure it's a concern on the serialization side, it's more the deserialization where we see recursive blowup.